presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/alluxio/AlluxioHiveMetastore.java

```diff
@@ -335,23 +338,22 @@ public void dropColumn(HiveIdentity identity, String databaseName, String tableN
     }
 
     @Override
-    public Optional<List<String>> getPartitionNames(HiveIdentity identity, String databaseName, String tableName)
+    public Optional<List<String>> getPartitionNamesByFilter(
+            HiveIdentity identity,
+            String databaseName,
+            String tableName,
+            List<String> columnNames,
+            TupleDomain<String> partitionKeysFilter)
     {
-        throw new PrestoException(NOT_SUPPORTED, "getPartitionNames");
-    }
+        if (partitionKeysFilter.isNone()) {
+            return Optional.of(Collections.emptyList());
+        }
+        Optional<List<String>> optionalParts = partitionKeyFilterToStringList(columnNames, partitionKeysFilter, false);
```

Comment:
We can remove all this client side filtering here since it's only a hint. So I think the whole method can be
```java
return ProtoUtils.toPartitionInfoList(
        client.readTable(databaseName, tableName, Constraint.getDefaultInstance()));

Reply:
got it. it looks like I need a mapping of `PartitionInfo::getPartitionName` as well